src/coreclr/nativeaot/Runtime/inc/MethodTable.h

```diff
@@ -133,6 +133,7 @@ class MethodTable
         };
     };
 
+    // used for component size for strings and arrays, type arg count for typedefs, otherwise holds FlagsEx bits
     uint16_t              m_usComponentSize;
     uint16_t              m_usFlags;
```

Comment:
We made this a single int on the managed side and made emission also happen as a single int. Should we collapse to a single integer here too?

Reply:
The native changes are coming shortly.

Reply:
It is not a lot, but we can optimize important things like computing object size in GC.